department-of-veterans-affairs/abd-vro
# enable kafka broker SSL config and spring kafka annotation

## What was the problem?
- At present, there's no system in place for the VRO platform to receive BIE Kafka topic messages across different environments.

Associated tickets or Slack threads:
- #1682 

## How does this fix it?[^1]
- The code allows the svc-bie-kafka microservice to establish a connection with the BIE Kafka broker and consume messages from five different topics related to contention classification.

## How to test this PR
- Presently, the Kafka consumer code, which is configured with SSL, cannot be tested locally due to the need for VA network access. Nevertheless, a subsequent PR could introduce a potential mock Kafka producer method. This, coupled with the mock Kafka broker, might provide a solution for testing the existing code.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
